File: backend/src/main/java/com/hotel_bidding/backend/service/UserManagementService.java
package com.hotel_bidding.backend.service;

import com.hotel_bidding.backend.dto.ApprovalActionRequest;
import com.hotel_bidding.backend.dto.PendingApprovalDTO;
import com.hotel_bidding.backend.dto.UserManagementStatsDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Map;

public interface UserManagementService {
    
    /**
     * Get user management statistics
     */
    UserManagementStatsDTO getUserManagementStats();
    
    /**
     * Get all pending approvals (both hotels and DMCs)
     */
    Page<PendingApprovalDTO> getPendingApprovals(Pageable pageable);
    
    /**
     * Get pending approval details by ID and type
     */
    PendingApprovalDTO getPendingApprovalById(String id, String type);
    
    /**
     * Process approval action (approve/reject)
     */
    Map<String, Object> processApprovalAction(
            String id, 
            String type, 
            ApprovalActionRequest request,
            String adminId,
            String adminUsername
    );
}

File: frontend/src/services/userManagementService.js
import api from './api';

/**
 * User Management Service
 * Handles API calls for admin user management
 */

const USER_MANAGEMENT_BASE_URL = '/admin/user-management';

/**
 * Get user management statistics
 */
export const getUserManagementStats = async () => {
  const response = await api.get(`${USER_MANAGEMENT_BASE_URL}/stats`);
  return response.data;
};

/**
 * Get all pending approvals
 */
export const getPendingApprovals = async (params = {}) => {
  const { page = 0, size = 10, sortBy = 'appliedDate', sortDirection = 'DESC' } = params;
  const response = await api.get(`${USER_MANAGEMENT_BASE_URL}/pending-approvals`, {
    params: { page, size, sortBy, sortDirection }
  });
  return response.data;
};

/**
 * Get pending approval details by ID
 */
export const getPendingApprovalById = async (id, type) => {
  const response = await api.get(`${USER_MANAGEMENT_BASE_URL}/pending-approvals/${id}`, {
    params: { type }
  });
  return response.data;
};

/**
 * Process approval action (approve/reject)
 */
export const processApprovalAction = async (id, type, actionData) => {
  const response = await api.post(
    `${USER_MANAGEMENT_BASE_URL}/pending-approvals/${id}/action`,
    actionData,
    { params: { type } }
  );
  return response.data;
};

/**
 * Approve a pending request
 */
export const approveRequest = async (id, type, note = '') => {
  return processApprovalAction(id, type, {
    action: 'APPROVE',
    note
  });
};

/**
 * Reject a pending request
 */
export const rejectRequest = async (id, type, reason, note = '') => {
  if (!reason || reason.trim() === '') {
    throw new Error('Rejection reason is required');
  }
  return processApprovalAction(id, type, {
    action: 'REJECT',
    reason,
    note
  });
};

export default {
  getUserManagementStats,
  getPendingApprovals,
  getPendingApprovalById,
  processApprovalAction,
  approveRequest,
  rejectRequest
};
